main.nf

```diff
@@ -1929,6 +1934,236 @@ process UNICYCLER_QUAST {
+
+    input:
+    set val(sample), val(single_end), file(reads) from ch_kraken2_minia
+    output:
+    set val(sample), val(single_end), file("*scaffolds.fa") into ch_minia_overlap_scaffolds,
+                                                                 ch_minia_quast,
+                                                                 ch_minia_blast,
+                                                                 ch_minia_abacas,
+                                                                 ch_minia_plasmidid
+
+    script:
+    if (single_end) {
+        """
+        minia -kmer-size 51 -abundance-min 20 -in $reads -out ${sample}.k51.a20
+        mv ${sample}.k51.a20.contigs.fa ${sample}.scaffolds.fa
+        """
+    }
+    else {
+        inputFiles = reads.join("\n")
+        """
+        echo '$inputFiles' > input_files.txt
+        minia -kmer-size 51 -abundance-min 20 -in input_files.txt -out ${sample}.k51.a20
+        mv ${sample}.k51.a20.contigs.fa ${sample}.scaffolds.fa
+        """
+    }
+}
+
+/*
+ * STEP 6.3.1: Overlap scaffolds with minimap2
+ */
+process MINIA_OVERLAP {
+    tag "$sample"
+    label 'process_medium'
+    publishDir "${params.outdir}/assembly/minia", mode: params.publish_dir_mode
+
+    when:
+    !params.skip_assembly && 'minia' in assemblers
+
+    input:
+    set val(sample), val(single_end), file(fasta) from ch_minia_overlap_scaffolds
+    file reference from ch_fasta_minia_overlap_scaffolds.collect()
+    output:
+    set val(sample), file(fasta), file("${sample}.paf") into ch_minia_induce_graph
+
+    """
+    minimap2 -c -x asm20 $reference $fasta > ${sample}.paf
+    """
+}
+
+/*
+ * STEP 6.3.2: Induce assembly graph with seqwish
+ */
+process MINIA_INDUCE_GRAPH {
+    tag "$sample"
+    label 'process_medium'
+    publishDir "${params.outdir}/assembly/minia", mode: params.publish_dir_mode
+
+    when:
+    !params.skip_assembly && 'minia' in assemblers
+
+    input:
+    set val(sample), file(fasta), file(alignment) from ch_minia_induce_graph
+    file reference from ch_fasta_minia_induce_graph.collect()
+    output:
+    set val(sample), file("${sample}.gfa") into ch_minia_call_variants
+
+    """
+    cat $fasta $reference > ${sample}.withRef.fasta
+    seqwish -s ${sample}.withRef.fasta -p $alignment -g ${sample}.gfa
+    """
+}
+
+/*
+ * STEP 6.3.3: Polish assembly graph and call variants with vg
+ */
+process MINIA_CALL_VARIANTS {
+    tag "$sample"
+    label 'process_medium'
+    publishDir "${params.outdir}/assembly/minia", mode: params.publish_dir_mode
+
+    when:
+    !params.skip_assembly && 'minia' in assemblers
+
+    input:
+    set val(sample), file(graph) from ch_minia_call_variants
+    output:
+    set val(sample), file("${sample}.vg"), file("${sample}.xg"), file("${sample}.vcf") into ch_minia_variants
+
+    """
+    vg view -Fv $graph > ${sample}.vg
+    vg convert -x ${sample}.vg > ${sample}.xg
+    vg deconstruct -p NC_045512.2 ${sample}.xg > ${sample}.vcf

```

Comment:
Notice this hard-coded reference sequence accession `NC_045512.2` here, that will have to come out

Reply:
Yep. I have used this bit of code to get the prefix using bash for now:
https://github.com/drpatelh/nf-core-viralrecon/blob/124823d293e8806f5dd558851ab6c4654e21a8b9/main.nf#L1526

However, this wont work for multi-fasta files @ekg?